forte_wrapper/twitter/twittersearch_processor.py

```diff
@@ -0,0 +1,139 @@
+from forte.common.resources import Resources
+from forte.data.multi_pack import MultiPack
+from forte.processors.base import MultiPackProcessor
+from forte.data.data_pack import DataPack
+from ft.onto.base_ontology import Document
+
+__all__ = [
+    "TweetSearchProcessor",
+]
+
+
+class TweetSearchProcessor(MultiPackProcessor):
+    """
+    TweetSearchProcessor is designed to query tweets with Twitter API.
+    Tweets will be returned as datapacks in input multipack.
+    """
+    # pylint: disable=useless-super-delegation
+    def __init__(self):
+        super().__init__()
+
+    def initialize(self, resources: Resources, configs: Config):
+        super().initialize(resources, configs)
+
+    @classmethod
+    def default_configs(cls) -> Dict[str, Any]:
+        # pylint: disable=line-too-long
+        """This defines a basic config structure for TweetSearchProcessor.
+        For more details about the parameters, refer to
+        https://docs.tweepy.org/en/latest/api.html#tweepy.API.search_tweets
+        and
+        https://developer.twitter.com/en/docs/twitter-api/v1/tweets/search/api-reference/get-search-tweets
+
+        Returns:
+            dictionary with the default config for this processor.
+        Following are the keys for this dictionary:
+        - `"credential_file"`: defines the path of credential file needed for
+            Twitter API usage.
+        - `"tweet_items"`: defines the number of tweets returned by processor.
+        - `"lang"`: language, restricts tweets to the given language,
+            default is 'en'.
+        - `"date_since"`: restricts tweets created after the given date.
+        - `"result_type"`: defines what type of search results to receive.
+            The default is “recent.” Valid values include:
+            mixed : include both popular and real time results in the response
+            recent : return only the most recent results in the response
+            popular : return only the most popular results in the response
+        - `"query_pack_name"`: The query pack's name, default is "query".
+        - `"response_pack_name_prefix"`: the pack name prefix to be used
+                in response datapacks.
+        """
+        config = super().default_configs()
+        config.update({
+            "credential_file": "",
+            "tweet_items": 5,
+            "lang": "en",
+            "date_since": "2020-01-01",
+            "result_type": 'recent',
+            "query_pack_name": "query",
+            "response_pack_name_prefix": "passage"
+        })
+        return config
+
+    def _process(self, input_pack: MultiPack):
+        r"""Search using Twitter API to fetch tweets for a query.
+        This query should be contained in the input multipack with name
+        `self.config.query_pack_name`.
+        This method adds new packs to `input_pack` containing the retrieved
+        results. Each result is added as a `ft.onto.base_ontology.Document`.
```

Comment:
Something like this?

Each result is added as a new data pack, and a `ft.onto.base_ontology.Document` annotation is used to cover the whole document.